src/device.cpp

```diff
@@ -1196,11 +1196,15 @@ cl_int cvk_device::get_device_host_timer(cl_ulong* device_timestamp,
     return CL_SUCCESS;
 }
 
-cl_ulong cvk_device::device_timer_to_host(cl_ulong dev, cl_ulong sync_dev,
-                                          cl_ulong sync_host) const {
-    if (sync_host > sync_dev) {
-        return (sync_host - sync_dev) + dev;
+cl_ulong cvk_device::device_timer_to_host(cl_ulong dev) {
+    if (dev > m_sync_dev) {
+        if (get_device_host_timer(&m_sync_dev, &m_sync_host) != CL_SUCCESS) {
+            return dev;
```

Comment:
This means we just silently return (most likely) garbage data in case an error is encountered. We should propagate the error I think.

Reply:
Sure, I'll change that.